<a href="https://colab.research.google.com/github/Malak-Khasawneh/Supermarket-Sales/blob/main/Supermarket_Sales.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [29]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [30]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [31]:
df = pd.read_csv('/content/drive/MyDrive/dataset/supermarket_sales-Sheet1.csv')
df.head()

Invoice ID Branch       City Customer type  Gender  \
0  750-67-8428      A     Yangon        Member  Female   
1  226-31-3081      C  Naypyitaw        Normal  Female   
2  631-41-3108      A     Yangon        Normal    Male   
3  123-19-1176      A     Yangon        Member    Male   
4  373-73-7910      A     Yangon        Normal    Male   

             Product line  Unit price  Quantity   Tax 5%     Total       Date  \
0       Health and beauty       74.69         7  26.1415  548.9715   1/5/2019   
1  Electronic accessories       15.28         5   3.8200   80.2200   3/8/2019   
2      Home and lifestyle       46.33         7  16.2155  340.5255   3/3/2019   
3       Health and beauty       58.22         8  23.2880  489.0480  1/27/2019   
4       Sports and travel       86.31         7  30.2085  634.3785   2/8/2019   

    Time      Payment    cogs  gross margin percentage  gross income  Rating  
0  13:08      Ewallet  522.83                 4.761905       26.1415     9.1  
1  10:29         Cash   76.40                 4.761905        3.8200     9.6  
2  13:23  Credit card  324.31                 4.761905       16.2155     7.4  
3  20:33      Ewallet  465.76                 4.761905       23.2880     8.4  
4  10:37      Ewallet  604.17                 4.761905       30.2085     5.3

*Exploring the Null values*

In [32]:
df.isna().sum()

Invoice ID                 0
Branch                     0
City                       0
Customer type              0
Gender                     0
Product line               0
Unit price                 0
Quantity                   0
Tax 5%                     0
Total                      0
Date                       0
Time                       0
Payment                    0
cogs                       0
gross margin percentage    0
gross income               0
Rating                     0
dtype: int64

*Exploring Duplicate Values*

In [33]:
df.duplicated().sum()

0

In [34]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 17 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Invoice ID               1000 non-null   object 
 1   Branch                   1000 non-null   object 
 2   City                     1000 non-null   object 
 3   Customer type            1000 non-null   object 
 4   Gender                   1000 non-null   object 
 5   Product line             1000 non-null   object 
 6   Unit price               1000 non-null   float64
 7   Quantity                 1000 non-null   int64  
 8   Tax 5%                   1000 non-null   float64
 9   Total                    1000 non-null   float64
 10  Date                     1000 non-null   object 
 11  Time                     1000 non-null   object 
 12  Payment                  1000 non-null   object 
 13  cogs                     1000 non-null   float64
 14  gross margin percentage  

In [35]:
cat_col = [col for col in df.columns if df[col].dtype == 'object']
df[cat_col]=df[cat_col].astype('string')

*Change the datatype of Date and Time columns*

In [37]:
df['Date'] = pd.to_datetime(df['Date'])
df['Time'] = pd.to_datetime(df['Time'], format='%H:%M').dt.time

**Change the index and sort it **

In [38]:
#set 'Date' as the new index
df.set_index('Date',inplace=True)
df.sort_index(inplace=True)
df.head()

Invoice ID Branch       City Customer type  Gender  \
Date                                                              
2019-01-01  770-42-8960      B   Mandalay        Normal    Male   
2019-01-01  651-88-7328      A     Yangon        Normal  Female   
2019-01-01  530-90-9855      A     Yangon        Member    Male   
2019-01-01  182-52-7000      A     Yangon        Member  Female   
2019-01-01  133-14-7229      C  Naypyitaw        Normal    Male   

                   Product line  Unit price  Quantity  Tax 5%    Total  \
Date                                                                     
2019-01-01   Food and beverages       21.12         8   8.448  177.408   
2019-01-01  Fashion accessories       65.74         9  29.583  621.243   
2019-01-01   Home and lifestyle       47.59         8  19.036  399.756   
2019-01-01    Sports and travel       27.04         4   5.408  113.568   
2019-01-01    Health and beauty       62.87         2   6.287  132.027   

                Time  Payment    cogs  gross margin percentage  gross income  \
Date                                                                           
2019-01-01  19:31:00     Cash  168.96                 4.761905         8.448   
2019-01-01  13:55:00     Cash  591.66                 4.761905        29.583   
2019-01-01  14:47:00     Cash  380.72                 4.761905        19.036   
2019-01-01  20:26:00  Ewallet  108.16                 4.761905         5.408   
2019-01-01  11:43:00     Cash  125.74                 4.761905         6.287   

            Rating  
Date                
2019-01-01     6.3  
2019-01-01     7.7  
2019-01-01     5.7  
2019-01-01     6.9  
2019-01-01     5.0

*Lets Find the number of items sold in each Product line*

In [39]:
df['Product line'].value_counts()

Product line
Fashion accessories       178
Food and beverages        174
Electronic accessories    170
Sports and travel         166
Home and lifestyle        160
Health and beauty         152
Name: count, dtype: Int64

*Fashion Accessories had the largest number of items sold, Health and beauty has the least number of items sold.*

In [68]:
total_sales = df.groupby('Product line')['Total'].sum().sort_values(ascending=False)
total_sales

Product line
Food and beverages        56144.8440
Sports and travel         55122.8265
Electronic accessories    54337.5315
Fashion accessories       54305.8950
Home and lifestyle        53861.9130
Health and beauty         49193.7390
Name: Total, dtype: float64

*Food and beverages has the largest amount of sales, Health and beauty has the least.*

*Lets find the sales of each branch*

In [69]:
df['Total'].sum()

322966.749

In [40]:
df['Branch'].value_counts()

Branch
A    340
B    332
C    328
Name: count, dtype: Int64

*Branch **A (Yangon)** has the greatest number of sales then branch **B (Mandalay)** then **C (Naypyitaw)** with the least number of sales.*

In [41]:
df.describe()

Unit price     Quantity       Tax 5%        Total        cogs  \
count  1000.000000  1000.000000  1000.000000  1000.000000  1000.00000   
mean     55.672130     5.510000    15.379369   322.966749   307.58738   
std      26.494628     2.923431    11.708825   245.885335   234.17651   
min      10.080000     1.000000     0.508500    10.678500    10.17000   
25%      32.875000     3.000000     5.924875   124.422375   118.49750   
50%      55.230000     5.000000    12.088000   253.848000   241.76000   
75%      77.935000     8.000000    22.445250   471.350250   448.90500   
max      99.960000    10.000000    49.650000  1042.650000   993.00000   

       gross margin percentage  gross income      Rating  
count             1.000000e+03   1000.000000  1000.00000  
mean              4.761905e+00     15.379369     6.97270  
std               6.131498e-14     11.708825     1.71858  
min               4.761905e+00      0.508500     4.00000  
25%               4.761905e+00      5.924875     5.50000  
50%               4.761905e+00     12.088000     7.00000  
75%               4.761905e+00     22.445250     8.50000  
max               4.761905e+00     49.650000    10.00000

*Customers have an average of 5.51 items in their cart. the maximum number of item in ones cart is 10 and the minimum is 1*

*Rating average of products is 6.972*

*The cheapest product is **10.08**USD and the most expensive one is **99.96**USD*

*Lets change the name of 'cogs' column: The cogs (cost of goods sold) column is not referring to the cost of goods paid by the store to obtain items and stock their shelves, but rather the pre-tax subtotal price for all items in a customer's cart.*

In [44]:
df.rename(columns={'cogs': 'Pre-tax total'}, inplace = True)
df.head()

Invoice ID Branch       City Customer type  Gender  \
Date                                                              
2019-01-01  770-42-8960      B   Mandalay        Normal    Male   
2019-01-01  651-88-7328      A     Yangon        Normal  Female   
2019-01-01  530-90-9855      A     Yangon        Member    Male   
2019-01-01  182-52-7000      A     Yangon        Member  Female   
2019-01-01  133-14-7229      C  Naypyitaw        Normal    Male   

                   Product line  Unit price  Quantity  Tax 5%    Total  \
Date                                                                     
2019-01-01   Food and beverages       21.12         8   8.448  177.408   
2019-01-01  Fashion accessories       65.74         9  29.583  621.243   
2019-01-01   Home and lifestyle       47.59         8  19.036  399.756   
2019-01-01    Sports and travel       27.04         4   5.408  113.568   
2019-01-01    Health and beauty       62.87         2   6.287  132.027   

                Time  Payment  Pre-tax total  gross margin percentage  \
Date                                                                    
2019-01-01  19:31:00     Cash         168.96                 4.761905   
2019-01-01  13:55:00     Cash         591.66                 4.761905   
2019-01-01  14:47:00     Cash         380.72                 4.761905   
2019-01-01  20:26:00  Ewallet         108.16                 4.761905   
2019-01-01  11:43:00     Cash         125.74                 4.761905   

            gross income  Rating  
Date                              
2019-01-01         8.448     6.3  
2019-01-01        29.583     7.7  
2019-01-01        19.036     5.7  
2019-01-01         5.408     6.9  
2019-01-01         6.287     5.0

*Lets see what Payment methods do Customers use*

In [46]:
df['Payment'].unique()

<StringArray>
['Cash', 'Ewallet', 'Credit card']
Length: 3, dtype: string

*so, Cash, E-wallet, and Credit Card are the only 3 Payment methods used*

*Lets find out the number of customers using each method*

In [47]:
df['Payment'].value_counts()

Payment
Ewallet        345
Cash           344
Credit card    311
Name: count, dtype: Int64

*so E-wallet is the most used Payment method and Credit card is the least used*

*Lets find out the number of sales for each Gender*

In [48]:
df['Gender'].value_counts()

Gender
Female    501
Male      499
Name: count, dtype: Int64

*The Female customers make more purchases but not by a big difference*

*Lets Find the number of sales in each month*

In [52]:
monthly_sales = df.resample('M').size()
print("Sales per Month:\n", monthly_sales)

Sales per Month:
 Date
2019-01-31    352
2019-02-28    303
2019-03-31    345
Freq: ME, dtype: int64


<ipython-input-52-8abbe445d427>:1: FutureWarning: 'M' is deprecated and will be removed in a future version, please use 'ME' instead.
  monthly_sales = df.resample('M').size()


*Then **January** has the largest number of sales, **March** comes next and the month with the least number of sales is **February**!*

*Now lets see in these three months which day is the day with the largest number of Sales*

In [58]:
daily_sales = df.resample('D').size()
highest_sales_day = pd.to_datetime(daily_sales.idxmax())
highest_sales_day

Timestamp('2019-02-07 00:00:00')

*What day was that?*

In [61]:
day_name = highest_sales_day.strftime('%A')
day_name

'Thursday'

*How many sales were made on that day?*

In [60]:
sales_count = daily_sales.max()
sales_count

20

*Seems like Thursday 2019-02-07 had 20 sales which is the highest number of sales in these 3 months*

*Lets find the peak shopping hour*

In [66]:
df['Time'] = pd.to_datetime(df['Time'], format='%H:%M:%S')
df['Hour'] = df['Time'].dt.hour

hourly_sales = df['Hour'].value_counts().sort_index()

peak_hour = hourly_sales.idxmax()
peak_hour


19

*Looks like 7:00 PM is the peak hour!*

*Lets find the type of the customers*

In [62]:
df['Customer type'].value_counts()

Customer type
Member    501
Normal    499
Name: count, dtype: Int64

*501 of the customers are **Members**.*